wurst/_handles/Destructable.wurst

```diff
@@ -69,6 +69,78 @@ public function destructable.setOccluderHeight(real height)
 public function destructable.getName() returns string
 	return GetDestructableName(this)
 
+// Extensions
+/** used to open a destructable gate*/
+public function destructable.gateOpen()
+	if (GetDestructableLife(this) > 0)
+		KillDestructable(this)
+
+/** used to close a destructable gate*/
+public function destructable.gateClose()
+	this.gateOpen()
+	DestructableRestoreLife(this, GetDestructableMaxLife(this), true)
+	SetDestructableAnimation(this, "stand")
+
+/** used to close a destructable gate*/
+public function destructable.gateDestroy()
+	this.gateOpen()
+	SetDestructableAnimation(this, "death")
+
+/** used to toggles a destructable gate open close state*/
+public function destructable.gateToggle()
+	if GetDestructableLife(this) > 0
+		this.gateClose()
+	else
+		this.gateOpen()
+
+constant let ELEVATOR_HEIGHT_MIN = 1
+constant let ELEVATOR_HEIGHT_MAX = 3		
+/** used to move a destructable elevator one step up*/
+public function destructable.elevatorUp()
+	this.setElevatorHeight(this.elevatorGetHeight()+1)
+
+/** used to move a destructable elevator one step down*/
+public function destructable.elevatorDown()
+	this.setElevatorHeight(this.elevatorGetHeight()-1)
+
+/** used to move a destructable elevator one step down*/
+public function destructable.elevatorSetHeight(int height)
+	this.setElevatorHeight(height)
+
+/** used to move a destructable elevator one step down*/
+public function destructable.elevatorGetHeight() returns int
+	let height = ELEVATOR_HEIGHT_MIN + (this.getOccluderHeight() / bj_CLIFFHEIGHT).toInt()
+	return height < ELEVATOR_HEIGHT_MIN or height > ELEVATOR_HEIGHT_MAX ? ELEVATOR_HEIGHT_MIN : height
```

Comment:
looks useful, don't throw this code away pls

Reply:
Sarcasm?

Reply:
no not sarcasm